---
title: "Downstream analysis 4 samples: LV20, 21 and PV20, 21"
author:
  - "trnguyen@ukaachen.de"
date: "Last update on `r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    number_sections: true
    df_print: paged
    toc: true
    toc_float:
      toc_collapsed: false
    toc_depth: 3
    theme: lumen
---
  
```{css zoom-lib-src, echo = FALSE}
script src = "https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"
```

```{js zoom-jquery, echo = FALSE}
$(document).ready(function() {
  $('body').prepend('<div class=\"zoomDiv\"><img src=\"\" class=\"zoomImg\"></div>');
  // onClick function for all plots (img's)
    $('img:not(.zoomImg)').click(function() {
      $('.zoomImg').attr('src', $(this).attr('src')).css({width: '100%'});
      $('.zoomDiv').css({opacity: '1', width: 'auto', border: '1px solid white', borderRadius: '5px', position: 'fixed', top: '50%', left: '50%', marginRight: '-50%', transform: 'translate(-50%, -50%)', boxShadow: '0px 0px 50px #888888', zIndex: '50', overflow: 'auto', maxHeight: '100%'});
    });
    // onClick function for zoomImg
    $('img.zoomImg').click(function() {
      $('.zoomDiv').css({opacity: '0', width: '0%'}); 
    });
  });
```

# Note
In this analysis, we examine 4 samples: LV20, LV21, PV20 and PV21. Raw count matrices and barcodes are generated by the `CellRanger` pipeline. 

Cells having the percentage of reads mapped to mitochondrial genes > 10% or having the percentage of ambient RNA contamination > 50% are removed. Doublet cells are detected by `DoubletFinder` and removed. 

**Some important sections:**

 - Section 1, 2, 3: Some quality control metrics.

 - Section 4: Expression of hashtag antibodies illustrated by Ridge plots and Violin plots.
 
 - Section 5: Cluster marker genes: Highly expressed marker genes in each cluster.
 
 - Section 6: Annotate clusters and show on TSNE. Table and bar plot show the abundance of each cell type in each sample. 
 
 - Section 8, 9, 10: Differential gene expression analysis for 3 cases:
 
    - All LV samples vs. all PV samples
    
    - LV20 vs PV20
    
    - LV21 vs PV21.
    
- Section 11, 12, 13: Focus on DC1, DC2, DC3 and NCM; highlight them on TSNE, heatmaps illustrating some marker genes and violin plots for the expression of hashtag antibodies.

- Section 14: Volcano plot for DE analysis of "All LV vs all PV", "LV20 vs PV20", "LV21 vs PV21".


**Note on the .html report: Click on the image to zoom-in and one more click to go back to the original**

```{r echo=FALSE, warning=FALSE, message=FALSE, results='hide', fig.height=10, fig.width=18}
##### clean up #####
gc()
rm(list = ls())

path.to.project.src <- "/home/hieunguyen/CRC1382/src_2023/AGBerres_SSchwarz"

source(file.path(path.to.project.src, "00_import_libraries.R"))
source(file.path(path.to.project.src, "00_helper_functions.R"))

path.to.src <- "/home/hieunguyen/CRC1382/src_2023/src_pipeline/scRNA_GEX_pipeline/processes_src"
source(file.path(path.to.src, "s8_integration_and_clustering.R"))

#####----------------------------------------------------------------------#####
# CONFIGURATIONS AND PREPRATIONS
#####----------------------------------------------------------------------#####
outdir <- "/home/hieunguyen/CRC1382/outdir"
path.to.main.input <- file.path(outdir, "AGBerres", "220311_Berres_MiedIII_scCITEseq")
s.obj <- readRDS(file.path(path.to.main.input, "s8_output", "AGBerres_4_samples_LV_PV.output.s8.rds"))
s.obj <- subset(s.obj, Doublet_classifications == "Singlet")

path.to.main.output <- file.path(path.to.main.input, "data_analysis")
dir.create(path.to.main.output, showWarnings = FALSE)

path.to.01.output <- file.path(path.to.main.output, "01_output_21092023")
dir.create(path.to.01.output, showWarnings = FALSE, recursive = TRUE)

cluster.markers <- readRDS(file.path(path.to.main.input, "s9_output", "AGBerres_4_samples_LV_PV.clusterMarkers.s9.rds"))
cluster.markers <- cluster.markers$diff.markers %>% subset(p_val_adj <= 0.05)
#####----------------------------------------------------------------------#####
# Transfering annotation from the previous analysis
#####----------------------------------------------------------------------#####
old.annotation <- read.csv(file.path(path.to.project.src, "old_cell_type_annotation.csv")) %>%
  column_to_rownames("X")

old.annotation <- subset(old.annotation, row.names(old.annotation) %in% colnames(s.obj))

meta.data <- s.obj@meta.data
merge.annotation <- merge(meta.data, old.annotation, by.x = 0, by.y = 0, all.x = TRUE)

s.obj <- AddMetaData(object = s.obj, metadata = merge.annotation$annotated.cell, col.name = "cell.annotation")

##### Add TSNE
s.obj <- RunTSNE(s.obj, reduction = "INTE_PCA", dims = 1:30, reduction.name="INTE_TSNE")
```

# Preprocessing and QC for GEX data
## Raw data Quality control  {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
for (plot.name in names(s.obj@misc$all.QC)){
 
    cat('###',plot.name,'{.unlisted .unnumbered} \n')
    
    ##### 
    # plots or tables that we want to show in tabs
    #####
    print(s.obj@misc$all.QC[plot.name])
    cat(' \n \n')
}
```


## Descriptive statistics and filtering threshold {.tabset}
This section is devoted to the descriptive statistics of the following varialbes: `nFeature_RNA, nCount_RNA, percent.mt, percent.ribo`. 

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
for (plot.item in c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent.ribo")){
  boxplot <- s.obj@meta.data %>% ggplot(aes_string(x = "name", y = plot.item)) +
    geom_boxplot(outlier.colour="black", outlier.shape=16,
               outlier.size=2, notch=FALSE) +
    ggtitle(sprintf("Boxplot: Distribution of %s in each dataset", plot.item))
  cat('###', plot.item,'{.unlisted .unnumbered} \n')
    
    ##### 
    # plots or tables that we want to show in tabs
    #####
  
  print(boxplot)
  
  cat(' \n \n')
}
```



## Descriptive statistics + TSNE {.tabset}

### % Mitochondrial
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
FeaturePlot(object = s.obj, reduction = "INTE_TSNE", feature = "percent.mt", label = TRUE, label.size = 8, pt.size = 0.5, label.color = "red", )
```

### % Ribosome
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
FeaturePlot(object = s.obj, reduction = "INTE_TSNE", feature = "percent.ribo", label = TRUE, label.size = 8, pt.size = 0.5, label.color = "red", )
```

### % nCount RNA
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
FeaturePlot(object = s.obj, reduction = "INTE_TSNE", feature = "nCount_RNA", label = TRUE, label.size = 8, pt.size = 0.5, label.color = "red", )
```

### % nGenes 
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
FeaturePlot(object = s.obj, reduction = "INTE_TSNE", feature = "nFeature_RNA", label = TRUE, label.size = 8, pt.size = 0.5, label.color = "red", )
```

## Cell cycle scoring {.tabset}

### Cell cycle, split by Phase
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(s.obj, reduction = "INTE_TSNE", split.by = "Phase")
```

### Cell cycle, group by Phase
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(s.obj, reduction = "INTE_TSNE", group.by = "Phase", label = TRUE, label.size = 8, pt.size = 0.5, label.box = TRUE, repel = TRUE)
```

### PCA, cell cycle, group by Phase
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
all.genes <- rownames(x = s.obj)
s.genes <- paste0("^", cc.genes$s.genes, "$", collapse = "|")
s.genes <- all.genes[grepl(s.genes, all.genes, ignore.case = TRUE)]
g2m.genes <- paste0("^", cc.genes$g2m.genes, "$", collapse = "|")
g2m.genes <- all.genes[grepl(g2m.genes, all.genes, ignore.case = TRUE)]
s.obj <- RunPCA(s.obj, features = c(s.genes, g2m.genes), nfeatures.print = 10, reduction.name="CELLCYCLED_PCA")

DimPlot(s.obj, reduction = "CELLCYCLED_PCA", group.by = "Phase", pt.size = 1)
```

# Dimension reduction with TSNE

## TSNE: all clusters 
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = s.obj, reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE) + 
  ggtitle(sprintf("TSNE: All clusters")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"), 
        title = element_text(size = 20, face = "bold"))
```

## TSNE: all samples
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = s.obj, reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE, group.by = "name") + 
  ggtitle(sprintf("TSNE Sample: All samples after integrating")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"), 
        title = element_text(size = 20, face = "bold"))
```

## TSNE: LV samples
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = subset(s.obj, stage == "LV"), reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE, group.by = "name") + 
  ggtitle(sprintf("TSNE Sample: All samples after integrating")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"), 
        title = element_text(size = 20, face = "bold"))
```

## TSNE: LV samples only, labeled by clusters
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = subset(s.obj, stage == "LV"), reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE, group.by = "seurat_clusters") + 
  ggtitle(sprintf("TSNE Sample: All samples after integrating")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"), 
        title = element_text(size = 20, face = "bold"))
```


## TSNE: PV samples
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object =  subset(s.obj, stage == "PV"), reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE, group.by = "name") + 
  ggtitle(sprintf("TSNE Sample: All samples after integrating")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"), 
        title = element_text(size = 20, face = "bold"))
```


## TSNE: PV samples only, labeled by clusters
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = subset(s.obj, stage == "PV"), reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE, group.by = "seurat_clusters") + 
  ggtitle(sprintf("TSNE Sample: All samples after integrating")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"), 
        title = element_text(size = 20, face = "bold"))
```

<!-- ## TSNE: cell type annotation (on the old results) -->
<!-- ```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12} -->
<!-- DimPlot(object = s.obj, reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE, group.by = "cell.annotation") + -->
<!--   ggtitle(sprintf("TSNE Sample: Clusters with cell type annotation")) + -->
<!--   theme(axis.text = element_text(size=20, face = "bold"), -->
<!--         axis.title = element_text(size=20, face = "bold"), -->
<!--         title = element_text(size = 20, face = "bold")) -->
<!-- ``` -->

### New clusters vs. manual cell-type annotation

- x-axis: old-cluster, cell annotation based on the first draft data. 

- y-axis: new-cluster

<!-- ```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12} -->
<!-- # subset(meta.data, select = c(seurat_clusters, cell.annotation)) %>% table() -->
<!-- ``` -->

<!-- ### New clusters vs. old clusters -->
<!-- ```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12} -->
<!-- check.old.new.clusters <- merge.annotation %>% subset(select = c(seurat_clusters.x, seurat_clusters.y)) -->
<!-- colnames(check.old.new.clusters) <- c("new.cluster", "old.cluster") -->
<!-- # check.old.new.clusters <- check.old.new.clusters %>% rowwise() %>%  -->
<!-- #   mutate(new.cluster = sprintf("new.cluster.%s", new.cluster)) %>% -->
<!-- #   mutate(old.cluster = sprintf("old.cluster%s", old.cluster)) -->

<!-- check.old.new.clusters %>% table() %>% as.data.frame() %>% pivot_wider(id_cols = "new.cluster",names_from = "old.cluster", values_from = "Freq") %>% column_to_rownames("new.cluster") %>% create_dt() -->
<!-- ``` -->

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
check.old.new.clusters <- merge.annotation %>% subset(select = c(seurat_clusters.x, annotated.cell))
colnames(check.old.new.clusters) <- c("new.cluster", "cell.annotation")
# check.old.new.clusters <- check.old.new.clusters %>% rowwise() %>% 
#   mutate(new.cluster = sprintf("new.cluster.%s", new.cluster)) %>%
#   mutate(old.cluster = sprintf("old.cluster%s", old.cluster))

check.old.new.clusters %>% table() %>% as.data.frame() %>% pivot_wider(id_cols = "new.cluster",names_from = "cell.annotation", values_from = "Freq") %>% column_to_rownames("new.cluster") %>% create_dt()
```

# Hashtag-Antibodies data (ADT)
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DefaultAssay(s.obj) <- "ADT"

counts.adt <- GetAssayData(s.obj, slot = "counts", assay = "ADT")
s.obj.HTO <- CreateSeuratObject(counts = counts.adt, assay = "HTO")

s.obj.HTO <- NormalizeData(s.obj.HTO, normalization.method = "CLR")

```

## Histogram: Raw count number of hashtags per cell {.tabset}
Histogram: Raw count number of hashtags per cell: 

- x-axis: Number of hashtags per cell

- y-axis: Number of cells

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
for (ht in unique(row.names(s.obj.HTO))){
  hashtag.counts <- counts.adt[ht, ] %>% as.data.frame()
  colnames(hashtag.counts) <- c("count")
  cat(sprintf("### Hashtag: %s \n", ht))
  p <- ggplot(subset(hashtag.counts, hashtag.counts$count >= 0), aes(x = count)) + geom_histogram(bins = max(hashtag.counts$count)) + ggtitle(ht) + 
    theme(text = element_text(size=20), 
          axis.text.x = element_text(hjust=1, size = 20),
          axis.text.y = element_text(hjust=1, size = 20)) 
  print(p)
  cat("\n \n")
}
```

## Histogram: log-transformed raw counts of hashtags per cell {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
for (ht in  unique(row.names(s.obj.HTO))){
  counts.adt.ht123 <- GetAssayData(s.obj.HTO, slot = "counts")
  hashtag.counts <- log2(counts.adt.ht123)[ht, ] %>% as.data.frame()
  colnames(hashtag.counts) <- c("count")
  cat(sprintf("### Hashtag: %s \n", ht))
  p <- ggplot(subset(hashtag.counts, hashtag.counts$count >= 0), aes(x = count)) + geom_histogram(bins = 10) + 
    ggtitle(ht)  + 
    theme(text = element_text(size=20), 
          axis.text.x = element_text(hjust=1, size = 20),
          axis.text.y = element_text(hjust=1, size = 20)) 
  print(p)
  cat("\n \n")
}
```

## Assigned cut-off 
```{r echo=FALSE, warning=FALSE, results='hide', message=FALSE}
s.obj.HTO <- HTODemux(s.obj.HTO, assay = "HTO", positive.quantile = 0.99)
```

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
Idents(s.obj.HTO) <- factor(x = Idents(s.obj), levels = levels(s.obj))
```

## Illustration by Ridge plots, Hashtag-Antibodies expression values in each cluster {.tabset}

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=12}
for (hashtag in rownames(s.obj.HTO[["HTO"]])){
  cat(sprintf("### %s \n", hashtag))
  p <- RidgePlot(s.obj.HTO, assay = "HTO", features = hashtag) +
    theme(legend.position = "none")
  print(p)
  cat("\n \n")
}
```


## Illustration by Violin plot {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=12}
for (hashtag in rownames(s.obj.HTO[["HTO"]])){
  cat(sprintf("### %s \n", hashtag))
  p <- VlnPlot(s.obj.HTO, assay = "HTO", features = c(hashtag)) + theme(legend.position = "none")
  print(p)
  cat("\n \n")
}
```

# Cluster marker genes

Identify differentially expressed genes in each cluster. 

## Feature plot {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
for (cluster.id in sort(unique(s.obj@meta.data$seurat_clusters))){
  DefaultAssay(s.obj) <- "RNA"
  cat(sprintf("### Cluster %s \n", cluster.id))
  tmp.cluster.markers <- subset(cluster.markers, cluster.markers$cluster == cluster.id) %>% arrange(desc(avg_log2FC))
  p <- FeaturePlot(object = s.obj, reduction = "INTE_TSNE", features = head(tmp.cluster.markers, 12)$gene, ncol = 3, label = TRUE, pt.size = 0.5, label.size = 5, label.color = "red")  
  print(p)
  cat("\n \n")
}
```

## Full tables of DE genes {.tabset}
```{r echo=FALSE, results='asis', include=FALSE}
cluster.markers %>% create_dt()
```

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
all.cluster.ids <- sort(unique(s.obj$seurat_clusters))
for (cluster.id in all.cluster.ids){
  tmp.table <- subset(cluster.markers, cluster.markers$cluster == cluster.id) %>% arrange(desc(avg_log2FC)) %>% rownames_to_column("Gene")
  cat(paste("\n\n### Cluster: ", cluster.id, "##\n"))
  print( htmltools::tagList(datatable(tmp.table, extensions = 'Buttons',
                                      filter = "top",
                                      options = list(dom = 'Blfrtip',
                                                     buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                                                     lengthMenu = list(c(10,25,50,-1),
                                                                       c(10,25,50,"All")),
                                                     columnDefs = list(list(
                                                       targets = "_all",
                                                       render = JS(
                                                         "function(data, type, row, meta) {",
                                                         "return type === 'display' && data != null && data.length > 100 ?",
                                                         "'<span title=\"' + data + '\">' + data.substr(0, 100) + '...</span>' : data;",
                                                         "}")
                                                     ))
                                      ))))
  cat("\n \n")  
}
```

# Cell annotation

We rename the cluster to our annotation

## Number of each cell type in a sample, table
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
cell.annotation <- c( `0`= "MoMac_0",
                      `1`= "Momac_1",
                      `2`= "MoMac_2",
                      `3`= "DC3",
                      `4`= "NCM",
                      `5`= "DC1",
                      `6`= "MoMac_6",
                      `7`= "MoMac_7",
                      `8`= "DC2",
                      `9`= "pDC",
                      `10`= "MDP 16+",
                      `11`= "MDP",
                      `12`= "MoMac_12",
                      `13`= "CDP",
                      `14`= "B-cell")

new.cluster.ids <- cell.annotation
names(new.cluster.ids) <- levels(s.obj)
s.obj <- RenameIdents(s.obj, new.cluster.ids)

cells <- hash()

all.cells <- Idents(s.obj)

for (cell.type in unique(all.cells)){
  cells[[cell.type]] <- names(all.cells[all.cells == cell.type])
}

all.LV.vs.all.PV <- hash()
LV20.vs.PV20 <- hash()
LV21.vs.PV21 <- hash()

count.cells.df <- data.frame(cell.annotation) %>% rownames_to_column("Cluster") %>% rowwise() %>% mutate(num.cells = length(cells[[cell.annotation]])) 
count.cells.df <- count.cells.df %>% rowwise() %>% 
  mutate(num.cells.LV20 = length(to_vec(for (item in cells[[cell.annotation]]) if (grepl("LV20", item) == TRUE) item))) %>%
  mutate(num.cells.LV21 = length(to_vec(for (item in cells[[cell.annotation]]) if (grepl("LV21", item) == TRUE) item))) %>%
  mutate(num.cells.PV20 = length(to_vec(for (item in cells[[cell.annotation]]) if (grepl("PV20", item) == TRUE) item))) %>%
  mutate(num.cells.PV21 = length(to_vec(for (item in cells[[cell.annotation]]) if (grepl("PV21", item) == TRUE) item)))

count.cells.df %>% create_dt()
```

# TSNE: cell annotation
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = s.obj, reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE) +
  ggtitle(sprintf("TSNE Sample: Clusters with cell type annotation")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"),
        title = element_text(size = 20, face = "bold"))
```

# TSNE: cell annotation, LV samples only
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = subset(s.obj, stage == "LV"), reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE) +
  ggtitle(sprintf("TSNE Sample: Clusters with cell type annotation")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"),
        title = element_text(size = 20, face = "bold"))
```

# TSNE: cell annotation, PV samples only
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DimPlot(object = subset(s.obj, stage == "PV"), reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, label.size = 10, pt.size = 1, repel = TRUE) +
  ggtitle(sprintf("TSNE Sample: Clusters with cell type annotation")) +
  theme(axis.text = element_text(size=20, face = "bold"),
        axis.title = element_text(size=20, face = "bold"),
        title = element_text(size = 20, face = "bold"))
```

If the table is empty, there is no significantly differentially expressed gene found.

# Differential gene expression analysis all LV vs all PV {.tabset}

We perform differential gene expression analysis between all two LV vs. all two PV samples in all clusters

**Note:**

- **Positive** `avg_log2FC` indicates that the gene is highly expressed in `LV` samples. 

- **Negative** `avg_log2FC` indicates that the gene is highly expressed in `PV` samples.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
if (file.exists(file.path(path.to.01.output, "all.LV.vs.all.PV.rds")) == FALSE){
  for (input.celltype in names(cells)){
    s.obj.at.cluster <- subset(s.obj, cells = cells[[input.celltype]])
    tmp.de.markers <- FindMarkers(object = s.obj.at.cluster, assay = "RNA", ident.1 = "LV", ident.2 = "PV", group.by = "stage")
    all.LV.vs.all.PV[[input.celltype]] <- tmp.de.markers
  }
  saveRDS(all.LV.vs.all.PV, file.path(path.to.01.output, "all.LV.vs.all.PV.rds"))  
} else {
  all.LV.vs.all.PV <- readRDS(file.path(path.to.01.output, "all.LV.vs.all.PV.rds"))
}

```


```{r echo=FALSE, results='asis', include=FALSE}
all.LV.vs.all.PV$CDP %>% rownames_to_column("Gene") %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```


```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
for (cluster.id in names(all.LV.vs.all.PV)){
  tmp.table <- all.LV.vs.all.PV[[cluster.id]] %>% 
    subset(p_val_adj <= 0.05) %>% 
    rownames_to_column("Gene") %>% 
    mutate_if(is.numeric, round, 6)
  cat(paste("\n\n### Cluster: ", cluster.id, "##\n"))
  print( htmltools::tagList(datatable(tmp.table, extensions = 'Buttons',
                                      filter = "top",
                                      options = list(dom = 'Blfrtip',
                                                     buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                                                     lengthMenu = list(c(10,25,50,-1),
                                                                       c(10,25,50,"All")),
                                                     columnDefs = list(list(
                                                       targets = "_all",
                                                       render = JS(
                                                         "function(data, type, row, meta) {",
                                                         "return type === 'display' && data != null && data.length > 100 ?",
                                                         "'<span title=\"' + data + '\">' + data.substr(0, 100) + '...</span>' : data;",
                                                         "}")
                                                     ))
                                      ))))
  cat("\n \n")  
}
```

# Differential gene expression analysis LV20 vs PV20 {.tabset}

We perform differential gene expression analysis between all two LV vs. all two PV samples in all clusters

**Note:**
  
- **Positive** `avg_log2FC` indicates that the gene is highly expressed in `LV` samples. 

- **Negative** `avg_log2FC` indicates that the gene is highly expressed in `PV` samples.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
if (file.exists(file.path(path.to.01.output, "LV20.vs.PV20.rds")) == FALSE){
  for (input.celltype in names(cells)){
  s.obj.at.cluster <- subset(s.obj, cells = cells[[input.celltype]])
  s.obj.at.cluster <- subset(s.obj.at.cluster, name %in% c("LV20", "PV20"))
  
  tmp.de.markers <- FindMarkers(object = s.obj.at.cluster, assay = "RNA", ident.1 = "LV", ident.2 = "PV", group.by = "stage")
  LV20.vs.PV20[[input.celltype]] <- tmp.de.markers
  }
  saveRDS(LV20.vs.PV20, file.path(path.to.01.output, "LV20.vs.PV20.rds"))
} else {
  LV20.vs.PV20 <- readRDS(file.path(path.to.01.output, "LV20.vs.PV20.rds"))
}


```


```{r echo=FALSE, results='asis', include=FALSE}
LV20.vs.PV20$CDP %>% rownames_to_column("Gene") %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```


```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
for (cluster.id in names(LV20.vs.PV20)){
  tmp.table <- LV20.vs.PV20[[cluster.id]] %>% 
    subset(p_val_adj <= 0.05) %>% 
    rownames_to_column("Gene") %>% 
    mutate_if(is.numeric, round, 6)
  cat(paste("\n\n### Cluster: ", cluster.id, "##\n"))
  print( htmltools::tagList(datatable(tmp.table, extensions = 'Buttons',
                                      filter = "top",
                                      options = list(dom = 'Blfrtip',
                                                     buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                                                     lengthMenu = list(c(10,25,50,-1),
                                                                       c(10,25,50,"All")),
                                                     columnDefs = list(list(
                                                       targets = "_all",
                                                       render = JS(
                                                         "function(data, type, row, meta) {",
                                                         "return type === 'display' && data != null && data.length > 100 ?",
                                                         "'<span title=\"' + data + '\">' + data.substr(0, 100) + '...</span>' : data;",
                                                         "}")
                                                     ))
                                      ))))
  cat("\n \n")  
}
```

# Differential gene expression analysis LV21 vs PV21 {.tabset}

We perform differential gene expression analysis between all two LV vs. all two PV samples in all clusters

**Note:**
  
- **Positive** `avg_log2FC` indicates that the gene is highly expressed in `LV` samples. 

- **Negative** `avg_log2FC` indicates that the gene is highly expressed in `PV` samples.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
if (file.exists(file.path(path.to.01.output, "LV21.vs.PV21.rds")) == FALSE){
  for (input.celltype in names(cells)){
  if (input.celltype != "B-cell"){
    s.obj.at.cluster <- subset(s.obj, cells = cells[[input.celltype]])
    s.obj.at.cluster <- subset(s.obj.at.cluster, name %in% c("LV21", "PV21"))
    
    tmp.de.markers <- FindMarkers(object = s.obj.at.cluster, assay = "RNA", ident.1 = "LV", ident.2 = "PV", group.by = "stage")
    LV21.vs.PV21[[input.celltype]] <- tmp.de.markers    
  } else {
    LV21.vs.PV21[["B-cell"]] <- data.frame()
  }
}
saveRDS(LV21.vs.PV21, file.path(path.to.01.output, "LV21.vs.PV21.rds"))
} else {
  LV21.vs.PV21 <- readRDS(file.path(path.to.01.output, "LV21.vs.PV21.rds"))
}

```

```{r echo=FALSE, results='asis', include=FALSE}
LV21.vs.PV21$CDP %>% rownames_to_column("Gene") %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
for (cluster.id in names(LV21.vs.PV21)){
  if (cluster.id != "B-cell"){
   tmp.table <- LV21.vs.PV21[[cluster.id]] %>% 
    subset(p_val_adj <= 0.05) %>% 
    rownames_to_column("Gene") %>% 
    mutate_if(is.numeric, round, 6)    
  } else {
    tmp.table <- data.frame(status = c("no data available"))
  }

  cat(paste("\n\n### Cluster: ", cluster.id, "##\n"))
  print( htmltools::tagList(datatable(tmp.table, extensions = 'Buttons',
                                      filter = "top",
                                      options = list(dom = 'Blfrtip',
                                                     buttons = c('copy', 'csv', 'excel', 'pdf', 'print'),
                                                     lengthMenu = list(c(10,25,50,-1),
                                                                       c(10,25,50,"All")),
                                                     columnDefs = list(list(
                                                       targets = "_all",
                                                       render = JS(
                                                         "function(data, type, row, meta) {",
                                                         "return type === 'display' && data != null && data.length > 100 ?",
                                                         "'<span title=\"' + data + '\">' + data.substr(0, 100) + '...</span>' : data;",
                                                         "}")
                                                     ))
                                      ))))
  cat("\n \n")  
}
```

# TSNE with highlighted DC1, DC2, DC3 and NCM, the rest in gray
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DC1_cells <- WhichCells(object = s.obj, ident = "DC1")
DC2_cells <- WhichCells(object = s.obj, ident = "DC2")
DC3_cells <- WhichCells(object = s.obj, ident = "DC3")
NCM_cells <- WhichCells(object = s.obj, ident = "NCM")

tsne_dimplot_with_celltype.gray <- DimPlot(s.obj, reduction = "INTE_TSNE", cells.highlight= list(DC1_cells, DC2_cells, DC3_cells, NCM_cells), cols= "lightgray",
        label = TRUE, label.box = TRUE, repel = TRUE, pt.size = 1,
        cols.highlight = c("#f59089", "#7af5b4", "#6fa6ed", "#d0db39")) + theme(legend.position = "none")
tsne_dimplot_with_celltype.gray
```

# Heatmaps without CD274, CD200 in Regulatory markers

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DefaultAssay(s.obj) <- "RNA"
s.obj.LV <- subset(s.obj, stage == "LV")
s.obj.PV <- subset(s.obj, stage == "PV")

cluster_of_interest <- c("DC1", "DC2", "DC3", "NCM")

gene_of_interest <- hash()

##### set of genes to show on the heatmap, 2022
# gene_of_interest[["Maturation"]] <- c("CD40", "CD86", "RELB", "CD83")
# gene_of_interest[["Regulatory"]] <- c("FAS", "BIRC3", "MARCKSL1")
# gene_of_interest[["TLRs_and_Adaptors"]] <- c("TICAM1", "TLR8", "TLR7", "TLR6", "TLR5", "TLR4", "TLR2", "TLR1")
# gene_of_interest[["Migration"]] <- c("CCR7", "MYO1G", "CXCL16", "ADAM8", "ICAM1", "FSCN1", "MARCKS", "MARCKSL1")

##### new set of genes to show on the heatmap, 2025
gene_of_interest[["Maturation"]] <- c("CD40", "CD86", "RELB", "CD83", "LAMP3", "CD80", "Il12b")
gene_of_interest[["Regulatory"]] <- c("FAS", "BIRC3", "MARCKSL1",
                                      "PD-L1", "PD-L2", "TIM-3", "IDO1", "CD200", "CD274", "PDCD1LG2", "ALDH1A2")
gene_of_interest[["TLRs_and_Adaptors"]] <- c("TICAM1", "TLR8", "TLR7", "TLR6", "TLR5", "TLR4", "TLR2", "TLR1")
gene_of_interest[["Migration"]] <- c("CCR7", "MYO1G", "CXCL16", "ADAM8", "ICAM1", "FSCN1", "MARCKS", "MARCKSL1",
                                     "CCL17", "CCL19", "CCL22")
```

## List of genes to be shown on the heatmap
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
for (group.name in names(gene_of_interest)){
  print(sprintf("Gene group: %s \n", group.name))
  print(sprintf("List of genes: %s \n", paste(gene_of_interest[[group.name]], collapse = ", ")))
}
```


## Internal comparison between clusters in each LV, PV 2-sample 

### LV samples {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DefaultAssay(s.obj) <- "RNA"
for (group in names(gene_of_interest)){
  cat(sprintf("#### %s \n", group))
  lv.df <- data.frame(gene_of_interest[[group]])
  colnames(lv.df) <- c("Gene")
  for (chosen.cluster in cluster_of_interest){
    tmp.s.obj.lv <- subset(s.obj.LV, cells = WhichCells(object = s.obj.LV, ident = chosen.cluster))
    count.matrix <- as.data.frame(GetAssayData(tmp.s.obj.lv, slot = "data"))
    count.matrix <- count.matrix[gene_of_interest[[group]], ]
    countdf <- as.data.frame(rowMeans(count.matrix))
    colnames(countdf) <- c(chosen.cluster)
    countdf <- countdf %>% rownames_to_column("Gene")
    lv.df <- merge(lv.df, countdf, by.x = "Gene", by.y = "Gene")
  }
    lv.df <- lv.df %>% column_to_rownames("Gene")
    lv.df$total <- rowSums(lv.df)
    # lv.df <- subset(lv.df, lv.df$total != 0)
    lv.df <- subset(lv.df, select = -c(total))
    # input.to.heatmap.lv <- log2(lv.df/t(rowMeans(lv.df)) + 0.1)
    input.to.heatmap.lv <- (lv.df - t(rowMeans(lv.df)))/(rowSds(as.matrix(lv.df)))
     
    tmp.heatmap.lv <- input.to.heatmap.lv %>% 
      as.data.frame() %>%
      rownames_to_column("Gene") %>%
      pivot_longer(-c(Gene), names_to = "Cluster", values_to = "Expression") %>%
      mutate(Cluster= fct_relevel(Cluster,colnames(input.to.heatmap.lv))) %>%
      ggplot(aes(x=Cluster, y=Gene, fill=Expression)) + 
      geom_raster() + 
    scale_fill_gradient2(low="#1417f5", high="#f52314", guide="colorbar") +
      theme(text = element_text(size=25))
    print(tmp.heatmap.lv)
    cat("\n \n")
}
```

### PV samples {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DefaultAssay(s.obj) <- "RNA"
for (group in names(gene_of_interest)){
  cat(sprintf("#### %s \n", group))
  ##### PREPARE HEATMAP DATA FOR PV SAMPLES
  pv.df <- data.frame(gene_of_interest[[group]])#
  colnames(pv.df) <- c("Gene")
  for (chosen.cluster in cluster_of_interest){
    tmp.s.obj.pv <- subset(s.obj.PV, cells = WhichCells(object = s.obj.PV, ident = chosen.cluster))
    count.matrix <- as.data.frame(GetAssayData(tmp.s.obj.pv, slot = "data"))
    count.matrix <- count.matrix[gene_of_interest[[group]], ]
    countdf <- as.data.frame(rowMeans(count.matrix))
    colnames(countdf) <- c(chosen.cluster)
    countdf <- countdf %>% rownames_to_column("Gene")
    pv.df <- merge(pv.df, countdf, by.x = "Gene", by.y = "Gene")
  }
  pv.df <- pv.df %>% column_to_rownames("Gene")
  pv.df$total <- rowSums(pv.df)
  # pv.df <- subset(pv.df, pv.df$total != 0)
  pv.df <- subset(pv.df, select = -c(total))
  # input.to.heatmap.pv <- log2(pv.df/t(rowMeans(pv.df)) + 0.1)
  input.to.heatmap.pv <- (pv.df - t(rowMeans(pv.df)))/(rowSds(as.matrix(pv.df)))
  
  tmp.heatmap.pv <- input.to.heatmap.pv %>% 
    as.data.frame() %>%
    rownames_to_column("Gene") %>%
    pivot_longer(-c(Gene), names_to = "Cluster", values_to = "Expression") %>%
    mutate(Cluster= fct_relevel(Cluster,colnames(input.to.heatmap.pv))) %>%
    ggplot(aes(x=Cluster, y=Gene, fill=Expression)) + 
    geom_raster() + 
    scale_fill_gradient2(low="#1417f5", high="#f52314", guide="colorbar") +
    theme(text = element_text(size=25))
  print(tmp.heatmap.pv)
  cat("\n \n")
}

```


## Cross condition comparisons on heatmaps {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DefaultAssay(s.obj) <- "RNA"
for (group in names(gene_of_interest)){
  cat(sprintf("### %s \n", group))
  lv.df <- data.frame(gene_of_interest[[group]])
  colnames(lv.df) <- c("Gene")
  for (chosen.cluster in cluster_of_interest){
    tmp.s.obj.lv <- subset(s.obj.LV, cells = WhichCells(object = s.obj.LV, ident = chosen.cluster))
    count.matrix <- as.data.frame(GetAssayData(tmp.s.obj.lv, slot = "data"))
    count.matrix <- count.matrix[gene_of_interest[[group]], ]
    countdf <- as.data.frame(rowMeans(count.matrix))
    colnames(countdf) <- c(chosen.cluster)
    countdf <- countdf %>% rownames_to_column("Gene")
    lv.df <- merge(lv.df, countdf, by.x = "Gene", by.y = "Gene")
  }
    lv.df <- lv.df %>% column_to_rownames("Gene")
    lv.df$total <- rowSums(lv.df)
    # lv.df <- subset(lv.df, lv.df$total != 0)
    lv.df <- subset(lv.df, select = -c(total))
    
  pv.df <- data.frame(gene_of_interest[[group]])
  colnames(pv.df) <- c("Gene")
  for (chosen.cluster in cluster_of_interest){
    tmp.s.obj.pv <- subset(s.obj.PV, cells = WhichCells(object = s.obj.PV, ident = chosen.cluster))
    count.matrix <- as.data.frame(GetAssayData(tmp.s.obj.pv, slot = "data"))
    count.matrix <- count.matrix[gene_of_interest[[group]], ]
    countdf <- as.data.frame(rowMeans(count.matrix))
    colnames(countdf) <- c(chosen.cluster)
    countdf <- countdf %>% rownames_to_column("Gene")
    pv.df <- merge(pv.df, countdf, by.x = "Gene", by.y = "Gene")
    }
  pv.df <- pv.df %>% column_to_rownames("Gene")
  pv.df$total <- rowSums(pv.df)
  # pv.df <- subset(pv.df, pv.df$total != 0)
  pv.df <- subset(pv.df, select = -c(total))

  colnames(lv.df) <- to_vec(for (item in colnames(lv.df)) sprintf("LV_%s", item))
  colnames(pv.df) <- to_vec(for (item in colnames(pv.df)) sprintf("PV_%s", item))
  
  lv.df <- lv.df %>% rownames_to_column("Gene")
  pv.df <- pv.df %>% rownames_to_column("Gene")
  
  merge.lv.pv.df <- merge(pv.df, lv.df, by.x = "Gene", by.y = "Gene") %>%
    column_to_rownames("Gene")

  input.to.heatmap <- (merge.lv.pv.df - t(rowMeans(merge.lv.pv.df)))/(rowSds(as.matrix(merge.lv.pv.df)))
  tmp.heatmap.lv.pv <- input.to.heatmap %>% 
    as.data.frame() %>%
    rownames_to_column("Gene") %>%
    pivot_longer(-c(Gene), names_to = "Cluster", values_to = "Expression") %>%
    mutate(Cluster= fct_relevel(Cluster,colnames(input.to.heatmap))) %>%
    ggplot(aes(x=Cluster, y=Gene, fill=Expression)) + 
    geom_raster() + 
    scale_fill_gradient2(low="#1417f5", high="#f52314", guide="colorbar") +
        theme(text = element_text(size=25))
  
  input.to.heatmap.lv.pv.only.lv <- input.to.heatmap[, c("LV_DC1", "LV_DC2", "LV_DC3", "LV_NCM")]
  input.to.heatmap.lv.pv.only.pv <- input.to.heatmap[, c("PV_DC1", "PV_DC2", "PV_DC3", "PV_NCM")]
  
  tmp.heatmap.lv <- input.to.heatmap.lv.pv.only.lv %>% 
    as.data.frame() %>%
    rownames_to_column("Gene") %>%
    pivot_longer(-c(Gene), names_to = "Cluster", values_to = "Expression") %>%
    mutate(Cluster= fct_relevel(Cluster,colnames(input.to.heatmap.lv.pv.only.lv))) %>%
    ggplot(aes(x=Cluster, y=Gene, fill=Expression)) + 
    geom_raster() + 
    scale_fill_gradient2(low="#1417f5", high="#f52314", guide="colorbar") +
        theme(text = element_text(size=25))
  
  tmp.heatmap.pv <- input.to.heatmap.lv.pv.only.pv %>% 
    as.data.frame() %>%
    rownames_to_column("Gene") %>%
    pivot_longer(-c(Gene), names_to = "Cluster", values_to = "Expression") %>%
    mutate(Cluster= fct_relevel(Cluster,colnames(input.to.heatmap.lv.pv.only.pv))) %>%
    ggplot(aes(x=Cluster, y=Gene, fill=Expression)) + 
    geom_raster() + 
    scale_fill_gradient2(low="#1417f5", high="#f52314", guide="colorbar") +
        theme(text = element_text(size=25))
  print(tmp.heatmap.lv.pv)
  cat("\n \n")
  }
```

# Violin plot for Surface markers of DC1, DC2, DC3 and NCM cells {.tabset}

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=12}
DefaultAssay(s.obj) <- "ADT"
s.obj <- ScaleData(s.obj, assay = "ADT")
s.obj <- NormalizeData(s.obj, assay = "ADT", normalization.method = "CLR")
Idents(s.obj) <- factor(x = Idents(s.obj), levels = sort(levels(s.obj)))

for (chosen.adt in row.names(s.obj)){
  cat(sprintf("## %s \n", chosen.adt))
  tmp.s.obj.violin <- s.obj[, Idents(s.obj) %in% c("DC1", "DC2", "DC3", "NCM")]
  tmp.s.obj.violin$stage <- factor(x = tmp.s.obj.violin$stage, levels = c("PV", "LV"))
  tmp.violin.plot <- VlnPlot(object = tmp.s.obj.violin, 
                             features = chosen.adt, pt.size = 0.05, assay = "ADT", split.by = "stage")
  print(tmp.violin.plot)
  cat("\n \n")
}
```

# Volcano plots for differential gene expression analysis

## All LV vs all PV {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
DefaultAssay(s.obj) <- "RNA"
for (gene.to.plot in names(all.LV.vs.all.PV)){
  cat(sprintf("### %s cells \n", gene.to.plot))
  tmp <- all.LV.vs.all.PV[[gene.to.plot]] %>% rownames_to_column("Gene") %>% rowwise  %>%
      mutate(sig = ifelse(p_val_adj <= 0.05, "Significant", "Not significant"))
  genes_to_show_on_volcanoplot <- subset(tmp, tmp$sig == "Significant")$Gene
  if (length(genes_to_show_on_volcanoplot) != 0){
        tmp <- tmp %>% rowwise %>%
          mutate(Gene = ifelse(Gene %in% genes_to_show_on_volcanoplot, Gene, ""))
        volcano.plot <- ggplot(data=tmp, 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), col=sig, label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, Cell type: %s, Sample: LV vs PV", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
      } else{
        volcano.plot <- ggplot(data=tmp, 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), col=sig, label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, Cell type: %s, Sample: LV vs PV", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) 
      }
    print(volcano.plot)
    cat("\n \n")
}

```

## LV20 vs PV20 {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
DefaultAssay(s.obj) <- "RNA"
for (gene.to.plot in names(LV20.vs.PV20)){
  cat(sprintf("### %s cells \n", gene.to.plot))
  tmp <- LV20.vs.PV20[[gene.to.plot]] %>% rownames_to_column("Gene") %>% rowwise  %>%
      mutate(sig = ifelse(p_val_adj <= 0.05, "Significant", "Not significant"))
  genes_to_show_on_volcanoplot <- subset(tmp, tmp$sig == "Significant")$Gene
  if (length(genes_to_show_on_volcanoplot) != 0){
        tmp <- tmp %>% rowwise %>%
          mutate(Gene = ifelse(Gene %in% genes_to_show_on_volcanoplot, Gene, ""))
        volcano.plot <- ggplot(data=tmp, 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), col=sig, label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, Cell type: %s, Sample: LV20 vs PV20", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
      } else{
        volcano.plot <- ggplot(data=tmp, 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), col=sig, label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, Cell type: %s, Sample: LV20 vs PV20", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) 
      }
    print(volcano.plot)
    cat("\n \n")
}

```
 

## LV21 vs PV21 {.tabset}
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
DefaultAssay(s.obj) <- "RNA"
for (gene.to.plot in names(LV21.vs.PV21)){
  if (nrow(LV21.vs.PV21[[gene.to.plot]]) != 0){
    cat(sprintf("### %s cells \n", gene.to.plot))
    tmp <- LV21.vs.PV21[[gene.to.plot]] %>% rownames_to_column("Gene") %>% rowwise  %>%
        mutate(sig = ifelse(p_val_adj <= 0.05, "Significant", "Not significant"))
    genes_to_show_on_volcanoplot <- subset(tmp, tmp$sig == "Significant")$Gene
    if (length(genes_to_show_on_volcanoplot) != 0){
          tmp <- tmp %>% rowwise %>%
            mutate(Gene = ifelse(Gene %in% genes_to_show_on_volcanoplot, Gene, ""))
          volcano.plot <- ggplot(data=tmp, 
                                 aes(x=avg_log2FC, y=-log10(p_val_adj), col=sig, label=Gene)) + 
            geom_point() + 
            scale_color_manual(values=c("#c0d2f0", "#f28095")) +
            theme_minimal() +
            geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
            geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
            geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
            #geom_text() +
            ggtitle(sprintf("Volcano plot, Cell type: %s, Sample: LV21 vs PV21", gene.to.plot)) +
            theme_bw() + 
            theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
            geom_text_repel(max.overlaps = 10000,
                            segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
        } else{
          volcano.plot <- ggplot(data=tmp, 
                                 aes(x=avg_log2FC, y=-log10(p_val_adj), col=sig, label=Gene)) + 
            geom_point() + 
            scale_color_manual(values=c("#c0d2f0", "#f28095")) +
            theme_minimal() +
            geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
            geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
            geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
            #geom_text() +
            ggtitle(sprintf("Volcano plot, Cell type: %s, Sample: LV21 vs PV21", gene.to.plot)) +
            theme_bw() + 
            theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) 
        }
      print(volcano.plot)
      cat("\n \n")
      }
}
```


# Differential gene expression DC1 vs DC2

**Note**

- **Positive** `avg_log2FC` indicates that the gene is highly expressed in *DC1* cells. 

- **Negative** `avg_log2FC` indicates that the gene is highly expressed in *DC2* cells.


```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
s.obj.DC1_DC2 <- subset(s.obj, idents = c("DC1", "DC2"))
s.obj.LV.DC1_DC2 <- subset(s.obj.LV, idents = c("DC1", "DC2"))
s.obj.PV.DC1_DC2 <- subset(s.obj.PV, idents = c("DC1", "DC2"))

s.obj.LV20.DC1_DC2 <- subset(s.obj.LV.DC1_DC2, name == "LV20")
s.obj.PV20.DC1_DC2 <- subset(s.obj.PV.DC1_DC2, name == "PV20")

s.obj.LV21.DC1_DC2 <- subset(s.obj.LV.DC1_DC2, name == "LV21")
s.obj.PV21.DC1_DC2 <- subset(s.obj.PV.DC1_DC2, name == "PV21")

compare_DC1_DC2 <- hash()
compare_DC1_DC2_raw <- hash() # all genes including nonsignificant genes p.val > 0.05
```

## All LV and all PV samples

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["all_LV_and_all_PV"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["all_LV_and_all_PV"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["all_LV_and_all_PV"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["all_LV_and_all_PV"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in all LV and all PV samples", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```


## All LV samples

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.LV.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["all_LV_samples"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["all_LV_samples"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["all_LV_samples"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["all_LV_samples"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in all LV samples", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```


## LV20 sample

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.LV20.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["LV20"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["LV20"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["LV20"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["LV20"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in LV20", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```

## LV21 sample

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.LV21.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["LV21"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["LV21"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["LV21"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["LV21"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in LV21", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```

## All PV samples

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.PV.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["all_PV_samples"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["all_PV_samples"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["all_PV_samples"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["all_PV_samples"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in all PV samples", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```


## PV20 sample

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.PV20.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["PV20"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["PV20"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["PV20"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["PV20"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in PV20", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```

## PV21 sample

### Tables
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=15, fig.height=10}
tmp <- FindMarkers(object = s.obj.PV21.DC1_DC2, assay = "RNA", ident.1 = "DC1", ident.2 = "DC2")
compare_DC1_DC2[["PV21"]] <- subset(tmp, tmp$p_val_adj <= 0.05) %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2_raw[["PV21"]] <- tmp %>% rownames_to_column("Gene") %>%
  rowwise() %>% mutate(abs_avg_log2FC = abs(avg_log2FC)) %>% arrange(desc(abs_avg_log2FC))

compare_DC1_DC2[["PV21"]]  %>% mutate_if(is.numeric, round, 6) %>% create_dt()
```

### Volcano plot 

All genes shown on volcano plot are significantly differentially expressed.

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
volcano.plot <- ggplot(data=compare_DC1_DC2[["PV21"]], 
                               aes(x=avg_log2FC, y=-log10(p_val_adj), label=Gene)) + 
          geom_point() + 
          scale_color_manual(values=c("#c0d2f0", "#f28095")) +
          theme_minimal() +
          geom_vline(xintercept=c(0), col="#9a9fa6", linetype='dotted') +
          geom_vline(xintercept=c(-1, 1), col="#9a9fa6", linetype='dotted') +
          geom_hline(yintercept=-log10(0.05), col="#9a9fa6", linetype='dotted') +
          #geom_text() +
          ggtitle(sprintf("Volcano plot, DC1 vs DC2 in PV21", gene.to.plot)) +
          theme_bw() + 
          theme(plot.title = element_text(hjust=0.5, face="bold", size = 12)) +
          geom_text_repel(max.overlaps = 10000,
                          segment.colour = "black", fontface = 'bold', seed = 42, colour = "red", size = 6)
ggplotly(volcano.plot, tooltip = "all")
```

# Generate T-SNE with cell type annotations
```{r echo=FALSE, warning=FALSE, message=FALSE, results='hide', fig.width=12, fig.height=8}
DC1_cells <- WhichCells(object = s.obj, ident = "DC1")
DC2_cells <- WhichCells(object = s.obj, ident = "DC2")
DC3_cells <- WhichCells(object = s.obj, ident = "DC3")
NCM_cells <- WhichCells(object = s.obj, ident = "NCM")

tsne_dimplot_with_celltype.gray <- DimPlot(s.obj, reduction = "INTE_TSNE", cells.highlight= list(DC1_cells, DC2_cells, DC3_cells, NCM_cells), cols= "grey",
        label = TRUE, label.box = TRUE, repel = TRUE,
        cols.highlight = c("#f59089", "#7af5b4", "#6fa6ed", "#d0db39")) + theme(legend.position = "none")

tsne_dimplot_with_celltype.gray.LV <- DimPlot(subset(s.obj, stage == "LV"), reduction = "INTE_TSNE", cells.highlight= list(DC1_cells, DC2_cells, DC3_cells, NCM_cells), cols= "grey",
        label = TRUE, label.box = TRUE, repel = TRUE,
        cols.highlight = c("#f59089", "#7af5b4", "#6fa6ed", "#d0db39")) + theme(legend.position = "none") +
  ggtitle("LV samples")

tsne_dimplot_with_celltype.gray.PV <- DimPlot(subset(s.obj, stage == "PV"), reduction = "INTE_TSNE", cells.highlight= list(DC1_cells, DC2_cells, DC3_cells, NCM_cells), cols= "grey",
        label = TRUE, label.box = TRUE, repel = TRUE,
        cols.highlight = c("#f59089", "#7af5b4", "#6fa6ed", "#d0db39")) + theme(legend.position = "none") +
  ggtitle("PV samples")

ggsave(plot = tsne_dimplot_with_celltype.gray.PV + tsne_dimplot_with_celltype.gray.LV, 
       filename = file.path(path.to.01.output, "TSNE_with_cell_types_gray_background_split_LV_PV.png"), device = "png", dpi = 350, width = 12, height = 8)

```

```{r echo=FALSE, warning=FALSE, message=FALSE, results='hide', fig.width=12, fig.height=8}
##### Save objects

saveRDS(object = s.obj, file = file.path(path.to.01.output, "s_obj.annotated.integrated.tsne.rds"))

saveRDS(object = all.LV.vs.all.PV, file.path(path.to.01.output, "DE_markers_all_LV_vs_all_PV.tsne.rds"))

saveRDS(object = LV20.vs.PV20, file.path(path.to.01.output, "DE_markers_LV20_vs_PV20.tsne.rds"))

saveRDS(object = LV21.vs.PV21, file.path(path.to.01.output, "DE_markers_LV21_vs_PV21.tsne.rds"))

for (name in names(compare_DC1_DC2)){
  saveRDS(object = compare_DC1_DC2[[name]], file = file.path(path.to.01.output, sprintf("DE_markers_DC1_vs_DC2_in_%s.tsne.rds", name)))
}

saveRDS(object = compare_DC1_DC2, file = file.path(path.to.01.output, "DE_markers_DC1_vs_DC2_all_cases.tsne.rds"))

saveRDS(object = compare_DC1_DC2_raw, file = file.path(path.to.01.output, "DE_markers_DC1_vs_DC2_all_cases_raw.tsne.rds"))
```

# Update 02.11.2023 {.tabset}

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=12, fig.height=8}
all.genes <- row.names(s.obj)
check.genes <- list(
  `PD-L1` = c("STAT1", "IRF1", "IL10", "ZAP70", "CBL"),
  `IL-4` = c("IL4R"),
  `IL-12` = c("IL23", "IL27", "IL35", "IL12R", "CD30", "TNFRSF8", "IP10", "CXCL10", "JAK2", "TYK2"),
  `AXL` = c("BCL3", "PI3K-AKT-mTOR", "MEKERK", "NF-κB", "JAK", "JAK/STAT", "ADAM10", "ADAM17", "GAS6")
)

available.genes <- hash()
for (group in names(check.genes)){
  available.genes[[group]] <- c()
  for (gene in check.genes[[group]]){
    if (gene %in% all.genes == TRUE){
      print(sprintf("Group: %s, gene: %s is in the data", group, gene))
      available.genes[[group]] <- c(available.genes[[group]], gene)
    } else {
      print(sprintf("Group: %s, gene: %s is NOT in the data", group, gene))
    }
  }
}
```

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=20, fig.height=14}
DefaultAssay(s.obj) <- "RNA"
for (group in names(available.genes)){
  cat(sprintf("## Group %s \n", group))
  p <- VlnPlot(object = s.obj, features = available.genes[[group]], slot = "data", group.by = "cell.annotation", split.by = "stage") + ggtitle(group)  + theme(legend.position = "bottom")
  print(p)
  cat("\n \n")
}

```

# Update 01.2025

## Change annotations for clusters 10-13
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
convert.new.annotations <- list(
  "10" = "unspecified early progenitors (UEP)",
  "11" = "immature monocytes (IM)",
  "12" = "CDP",
  "13" = "MDP"
)

meta.data <- s.obj@meta.data %>% rownames_to_column("barcode") %>%
  rowwise() %>% 
  mutate(new.cell.annotation = ifelse(
    seurat_clusters %in% names(convert.new.annotations),
    convert.new.annotations[[sprintf("%s", seurat_clusters)]],
    cell.annotation
  )) %>%
  column_to_rownames("barcode")

meta.data <- meta.data[row.names(s.obj@meta.data), ]
s.obj <- AddMetaData(object = s.obj, metadata = meta.data$new.cell.annotation, col.name = "new.cell.annotation")

saveRDS(s.obj, file.path(path.to.01.output, "s_obj.annotated.integrated.tsne.NewAnnotation.rds"))
```

### New TSNE plot with new cell annotation
```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
DimPlot(object = s.obj, reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, repel = TRUE, group.by = "new.cell.annotation")
```

