---
title: "Downstream analysis 4 samples: LV20, 21 and PV20, 21, sub.clusters DC1, DC2, DC3"
author:
  - "trnguyen@ukaachen.de"
date: "Last update on `r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    number_sections: true
    df_print: paged
    toc: true
    toc_float:
      toc_collapsed: false
    toc_depth: 3
    theme: lumen
---
  
```{css zoom-lib-src, echo = FALSE}
script src = "https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"
```

```{js zoom-jquery, echo = FALSE}
$(document).ready(function() {
  $('body').prepend('<div class=\"zoomDiv\"><img src=\"\" class=\"zoomImg\"></div>');
  // onClick function for all plots (img's)
    $('img:not(.zoomImg)').click(function() {
      $('.zoomImg').attr('src', $(this).attr('src')).css({width: '100%'});
      $('.zoomDiv').css({opacity: '1', width: 'auto', border: '1px solid white', borderRadius: '5px', position: 'fixed', top: '50%', left: '50%', marginRight: '-50%', transform: 'translate(-50%, -50%)', boxShadow: '0px 0px 50px #888888', zIndex: '50', overflow: 'auto', maxHeight: '100%'});
    });
    // onClick function for zoomImg
    $('img.zoomImg').click(function() {
      $('.zoomDiv').css({opacity: '0', width: '0%'}); 
    });
  });
```

# Note
In this analysis, we examine 4 samples: LV20, LV21, PV20 and PV21. Raw count matrices and barcodes are generated by the `CellRanger` pipeline. 

Cells having the percentage of reads mapped to mitochondrial genes > 10% or having the percentage of ambient RNA contamination > 50% are removed. Doublet cells are detected by `DoubletFinder` and removed. 

**Some important sections:**

 - Section 1, 2, 3: Some quality control metrics.

 - Section 4: Expression of hashtag antibodies illustrated by Ridge plots and Violin plots.
 
 - Section 5: Cluster marker genes: Highly expressed marker genes in each cluster.
 
 - Section 6: Annotate clusters and show on TSNE. Table and bar plot show the abundance of each cell type in each sample. 
 
 - Section 8, 9, 10: Differential gene expression analysis for 3 cases:
 
    - All LV samples vs. all PV samples
    
    - LV20 vs PV20
    
    - LV21 vs PV21.
    
- Section 11, 12, 13: Focus on DC1, DC2, DC3 and NCM; highlight them on TSNE, heatmaps illustrating some marker genes and violin plots for the expression of hashtag antibodies.

- Section 14: Volcano plot for DE analysis of "All LV vs all PV", "LV20 vs PV20", "LV21 vs PV21".


**Note on the .html report: Click on the image to zoom-in and one more click to go back to the original**

```{r echo=FALSE, warning=FALSE, message=FALSE, results='hide', fig.height=10, fig.width=18}
##### clean up #####
gc()
rm(list = ls())

path.to.project.src <- "/home/hieunguyen/CRC1382/src_2023/AGBerres_SSchwarz"

source(file.path(path.to.project.src, "00_import_libraries.R"))
source(file.path(path.to.project.src, "00_helper_functions.R"))

path.to.src <- "/home/hieunguyen/CRC1382/src_2023/src_pipeline/scRNA_GEX_pipeline/processes_src"
source(file.path(path.to.src, "s8_integration_and_clustering.R"))

#####----------------------------------------------------------------------#####
# CONFIGURATIONS AND PREPRATIONS
#####----------------------------------------------------------------------#####
outdir <- "/home/hieunguyen/CRC1382/outdir"
path.to.main.input <- file.path(outdir, "AGBerres", "220311_Berres_MiedIII_scCITEseq")
path.to.main.output <- file.path(path.to.main.input, "data_analysis")
dir.create(path.to.main.output, showWarnings = FALSE)

path.to.01.output <- file.path(path.to.main.output, "01_output_21092023")
path.to.03.output <- file.path(outdir, "AGBerres", "human_dc_Maier_et_al_data", "03_output")
path.to.04.output <- file.path(path.to.main.output, "04_output")

s.obj <- readRDS(file.path(path.to.01.output, "s_obj.annotated.integrated.tsne.NewAnnotation.rds"))

s.obj.maier <- readRDS(file.path(path.to.03.output, "Maier_data_seurat_object.rds"))

subset.s.obj <- subset(s.obj, new.cell.annotation %in% c("DC1", "DC2", "DC3") )
```

```{r echo=FALSE, warning=FALSE, message=FALSE, results='asis', fig.width=14, fig.height=10}
DimPlot(object = s.obj, reduction = "INTE_TSNE", label = TRUE, label.box = TRUE, repel = TRUE, group.by = "new.cell.annotation")
```